### PR TITLE
Fixes #2714 Add rebuild support to os_server_actions

### LINKED ISSUE #2714: os_server_actions nova rebuild server
##### ISSUE TYPE

Feature Idea
##### COMPONENT NAME

os_server module
##### ANSIBLE VERSION

N/A
##### SUMMARY

Could it be possible to implement the nova rebuild server action in the os_server_actions module?

We use openstack in our CI pipeline, and this is far more cheaper then booting a new server every
time we test.

### PR DESCRIPTION
**Pull request by rcarrillocruz**
##### ISSUE TYPE
- Feature Pull Request
##### COMPONENT NAME

openstack/os_server_actions
##### ANSIBLE VERSION

```
ansible 2.2.0 (devel 6dc148d82c) last updated 2016/08/01 10:19:49 (GMT +000)
  lib/ansible/modules/core: (add_rebuild_os_server_action a005b5fe41) last updated 2016/08/01 11:11:36 (GMT +000)
  lib/ansible/modules/extras: (detached HEAD db979dde74) last updated 2016/08/01 10:19:53 (GMT +000)
  config file = 
  configured module search path = Default w/o overrides
```
##### SUMMARY

This PR adds the ability to rebuild an OpenStack VM.

```
vagrant@vagrant-ubuntu-trusty-64:~$ openstack --os-cloud devstack-admin server show testserver
+--------------------------------------+----------------------------------------------------------------+
| Field                                | Value                                                          |
+--------------------------------------+----------------------------------------------------------------+
| OS-DCF:diskConfig                    | MANUAL                                                         |
| OS-EXT-AZ:availability_zone          | nova                                                           |
| OS-EXT-SRV-ATTR:host                 | vagrant-ubuntu-trusty-64.localdomain                           |
| OS-EXT-SRV-ATTR:hypervisor_hostname  | vagrant-ubuntu-trusty-64.localdomain                           |
| OS-EXT-SRV-ATTR:instance_name        | instance-00000001                                              |
| OS-EXT-STS:power_state               | Running                                                        |
| OS-EXT-STS:task_state                | None                                                           |
| OS-EXT-STS:vm_state                  | active                                                         |
| OS-SRV-USG:launched_at               | 2016-08-01T10:29:01.000000                                     |
| OS-SRV-USG:terminated_at             | None                                                           |
| accessIPv4                           |                                                                |
| accessIPv6                           |                                                                |
| addresses                            | public=2001:db8::b, 172.24.4.4                                 |
| config_drive                         | True                                                           |
| created                              | 2016-08-01T10:28:17Z                                           |
| flavor                               | cirros256 (c1)                                                 |
| hostId                               | f19aa129dd9e87abc1b04f51b333a2dabb83aca29d0c28fdbceb221b       |
| id                                   | f28aaf9c-6d3f-401d-a28d-7592579eccaa                           |
| image                                | cirros-0.3.4-x86_64-uec (aa2dd71b-69dd-45ab-a979-81339e6755b1) |
| key_name                             | None                                                           |
| name                                 | testserver                                                     |
| os-extended-volumes:volumes_attached | []                                                             |
| progress                             | 0                                                              |
| project_id                           | 74de37d1b73149f18603a4520bcb70b0                               |
| properties                           |                                                                |
| security_groups                      | [{u'name': u'default'}]                                        |
| status                               | ACTIVE                                                         |
| updated                              | 2016-08-01T10:29:03Z                                           |
| user_id                              | ee979b97f6474e549f7eb884cb7dde92                               |
+--------------------------------------+----------------------------------------------------------------+
vagrant@vagrant-ubuntu-trusty-64:~$ ansible-playbook -i 'localhost,' test_os_server_rebuild.yml 

PLAY [localhost] ***************************************************************

TASK [os_server_actions] *******************************************************
changed: [localhost]

PLAY RECAP *********************************************************************
localhost                  : ok=1    changed=1    unreachable=0    failed=0   

vagrant@vagrant-ubuntu-trusty-64:~$ openstack --os-cloud devstack-admin server show testserver
+--------------------------------------+----------------------------------------------------------------+
| Field                                | Value                                                          |
+--------------------------------------+----------------------------------------------------------------+
| OS-DCF:diskConfig                    | MANUAL                                                         |
| OS-EXT-AZ:availability_zone          | nova                                                           |
| OS-EXT-SRV-ATTR:host                 | vagrant-ubuntu-trusty-64.localdomain                           |
| OS-EXT-SRV-ATTR:hypervisor_hostname  | vagrant-ubuntu-trusty-64.localdomain                           |
| OS-EXT-SRV-ATTR:instance_name        | instance-00000001                                              |
| OS-EXT-STS:power_state               | Running                                                        |
| OS-EXT-STS:task_state                | rebuild_spawning                                               |
| OS-EXT-STS:vm_state                  | active                                                         |
| OS-SRV-USG:launched_at               | 2016-08-01T10:29:01.000000                                     |
| OS-SRV-USG:terminated_at             | None                                                           |
| accessIPv4                           |                                                                |
| accessIPv6                           |                                                                |
| addresses                            | public=2001:db8::b, 172.24.4.4                                 |
| config_drive                         | True                                                           |
| created                              | 2016-08-01T10:28:17Z                                           |
| flavor                               | cirros256 (c1)                                                 |
| hostId                               | f19aa129dd9e87abc1b04f51b333a2dabb83aca29d0c28fdbceb221b       |
| id                                   | f28aaf9c-6d3f-401d-a28d-7592579eccaa                           |
| image                                | cirros-0.3.4-x86_64-uec (aa2dd71b-69dd-45ab-a979-81339e6755b1) |
| key_name                             | None                                                           |
| name                                 | testserver                                                     |
| os-extended-volumes:volumes_attached | []                                                             |
| progress                             | 0                                                              |
| project_id                           | 74de37d1b73149f18603a4520bcb70b0                               |
| properties                           |                                                                |
| security_groups                      | [{u'name': u'default'}]                                        |
| status                               | REBUILD                                                        |
| updated                              | 2016-08-01T11:09:51Z                                           |
| user_id                              | ee979b97f6474e549f7eb884cb7dde92                               |
+--------------------------------------+----------------------------------------------------------------+
```

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @rcarrillocruz for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by rcarrillocruz** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @rcarrillocruz. To the current maintainers, @emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/openstack/os_server_actions.py [hunk 1]
```diff
@@ -203,6 +215,16 @@ def main():
...
 
+        elif action == 'rebuild':
+            image = cloud.get_image(image)
+
+            if image is None:
+                module.fail_json(msg="Image does not exist")
+
+            # rebuild doesn't set a state, just do it
+            cloud.nova_client.servers.rebuild(server=server.id, image=image.id)
+            module.exit_json(changed=True)
```
Comment (omgjlk) [inline cloud/openstack/os_server_actions.py:229]: This should support waiting until the rebuild is complete.

**Comment by omgjlk** [comment]
This looks  okay, but it should support waiting for the rebuild to finish, just like it waits for the other actions to finish.

needs_revision

**Comment by gregdek** [comment]
Thanks @rcarrillocruz for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@rcarrillocruz A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by rcarrillocruz** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @rcarrillocruz. To the current maintainers, @emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by omgjlk** [comment]
shipit

**Comment by gregdek** [comment]
Thanks again to @rcarrillocruz for this PR, and thanks @emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request merged**